pkg/configs/client/client.go

```diff
@@ -163,3 +119,52 @@ func (c ConfigsResponse) GetLatestConfigID() configs.ID {
 	}
 	return latest
 }
+
+// ListAllRuleGroups polls the configdb server and returns the updated rule groups
+func (c *ConfigDBClient) ListAllRuleGroups(ctx context.Context) (map[string]rules.RuleGroupList, error) {
+	endpoint := fmt.Sprintf("%s/private/api/prom/configs/rules", c.URL.String())
+	response, err := doRequest(endpoint, c.Timeout, 0, "ListAllRuleGroups")
+	if err != nil {
+		return nil, err
+	}
+
+	configs := map[string]configs.VersionedRulesConfig{}
+	for id, view := range response.Configs {
+		cfg := view.GetVersionedRulesConfig()
+		if cfg != nil {
+			configs[id] = *cfg
+		}
+	}
+
+	newRules := map[string]rules.RuleGroupList{}
+
+	for user, cfg := range configs {
+		userRules := rules.RuleGroupList{}
+		if cfg.IsDeleted() {
```

Comment:
Is this like a diff of rules you're parsing and rebuilding the final state?

Reply:
Yes, that is essentially what is going on.